src/include/migraphx/op/pointwise.hpp

```diff
@@ -86,8 +81,15 @@ struct pointwise
                 [&](auto&& name, auto&& arg) { return std::make_pair(name, arg.element(i)); });
 
             auto results = run(pm, params);
-            assert(results.size() == 1);
-            visit_all(output, results.front())([&](auto out, auto x) { out[i] = x.front(); });
+            assert(results.size() == output.get_sub_objects().size() or
+                   (results.size() == 1 and output.get_sub_objects().empty()));
```

Comment:
Can there be a tuple with size 1 ?

Reply:
I think this handles the single input single output case still is how I interpreted it. Either you get type type and ensure your sub objects match or you have default behavior after a run

Reply:
> Can there be a tuple with size 1 ?

Yes but for single output we just return it directly instead of a tuple. This is so the single output will still work the same, some more work would be needed to handle tuple outputs in the fuse_pointwise(which outside of the scope of this PR).